modules/reindex/src/internalClusterTest/java/org/elasticsearch/migration/AbstractFeatureMigrationIntegTest.java

```diff
@@ -124,10 +124,16 @@ public void setupTestPlugin() {
     }
 
     public <T extends Plugin> T getPlugin(Class<T> type) {
+        createMasterNode();
         final PluginsService pluginsService = internalCluster().getCurrentMasterNodeInstance(PluginsService.class);
         return pluginsService.filterPlugins(type).stream().findFirst().get();
     }
 
+    protected void createMasterNode() {
+        // most tests do not care how master node is created, and can rely on internalCluster().getCurrentMasterNodeInstance
+        // but some like SystemIndexMigrationIT need master node to be master only node
```

Comment:
Shouldn't this have a default implementation which creates a master node?

Reply:
I know it looks odd. Maybe this method could return a pluginsService instead?
Let me update